File: src/Cpp/Solution03.cpp
#include<iostream>
#include<vector>
using namespace std;

class Solution {
public:
    int findRepeatNumber(vector<int>& nums) {
        int len = nums.size();
        int temp = 0;
        for (int i = 0; i < len; ++i) {
            if (nums[i] != i) {
                if (nums[i] == nums[nums[i]]) {
                    return nums[i];
                } else {
                    temp = nums[i];
                    nums[i] = nums[temp];
                    nums[temp] = temp;
                }
            }
        }

        return -1;
    }
};

int main() {
    auto s = new Solution();
    vector<int> v = {2, 3, 1, 0, 2, 5, 3};
    cout << s->findRepeatNumber(v) << endl;
    delete s;
}

File: src/Python/Solution03.py
from __future__ import annotations
from typing import List

class Solution:
    def findRepeatNumber(self, nums: List[int]) -> int:
        size = len(nums)
        temp = 0
        for index, value in enumerate(nums):
            if value != index:
                if value == nums[value]:
                    return value
                else:
                    temp = value
                    value = nums[temp]
                    nums[temp] = temp
        return -1

if __name__ == "__main__":
    s = Solution()
    print(s.findRepeatNumber([2, 3, 1, 0, 2, 5, 3]))
